cmd/git-sync/main.go

```diff
@@ -688,6 +688,12 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 		return err
 	}
 
+	// Check for rev after fetch
+	_, err = localHashForRev(ctx, rev, gitRoot)
+	if err != nil {
+		return false, "", err
+	}
+	
 	// GC clone
```

Comment:
We can remove the worktree if adding a worktree fails below (L707), by calling `os.RemoveAll(worktreePath)`. That will fix #411.

Reply:
https://github.com/kubernetes/git-sync/pull/414

Reply:
At this point the worktree has not been created yet.

Reply:
I can't make a comment at a section with no changes. This comment is about L707 below (after the worktree is created).

Reply:
Fix #414 looks good to me